scaleway/resource_instance_server.go

```diff
@@ -208,7 +214,7 @@ func resourceScalewayInstanceServerCreate(d *schema.ResourceData, m interface{})
 		CommercialType:    commercialType,
 		EnableIPv6:        d.Get("enable_ipv6").(bool),
 		SecurityGroup:     expandID(d.Get("security_group_id")),
-		DynamicIPRequired: Bool(false),
+		DynamicIPRequired: Bool(!d.Get("disable_public_ip").(bool)),
```

Comment:
why not using scw.BoolPtr ?

Reply:
Helpers exist here as well, let's refacto this in another PR?